From fc1fe4649dc62ecfffc94b76d908d66905d52e16 Mon Sep 17 00:00:00 2001
From: Pavithra R <quic_pavir@quicinc.com>
Date: Wed, 6 Mar 2024 22:29:41 +0530
Subject: [PATCH] arm64: dts: qcom: Add EDMA node for IPQ9574

Add EDMA (Ethernet DMA) device tree node for IPQ9574 to
enable ethernet support.

Change-Id: I87d7c50f2485c8670948dce305000337f6499f8b
Signed-off-by: Pavithra R <quic_pavir@quicinc.com>
Alex G: Add missing edma interrupts
Signed-off-by: Alexandru Gagniuc <mr.nuke.me@gmail.com>
---
 arch/arm64/boot/dts/qcom/ipq9574.dtsi | 139 ++++++++++++++++++++++++++
 1 file changed, 139 insertions(+)

--- a/arch/arm64/boot/dts/qcom/ipq9574.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq9574.dtsi
@@ -1314,6 +1314,145 @@
 					     "nssnoc_memnoc",
 					     "memnoc_nssnoc",
 					     "memnoc_nssnoc_1";
+
+			ethernet-dma {
+				clocks = <&nsscc NSS_CC_PPE_EDMA_CLK>,
+					 <&nsscc NSS_CC_PPE_EDMA_CFG_CLK>;
+				clock-names = "sys",
+					      "apb";
+				resets = <&nsscc EDMA_HW_RESET>;
+				reset-names = "edma";
+				interrupts = <GIC_SPI 363 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 364 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 365 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 366 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 367 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 368 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 369 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 370 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 371 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 372 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 373 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 374 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 375 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 376 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 377 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 378 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 379 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 380 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 381 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 382 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 383 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 384 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 509 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 508 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 507 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 506 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 505 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 504 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 503 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 502 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 501 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 500 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 355 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 356 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 357 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 358 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 359 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 360 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 361 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 362 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 331 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 332 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 333 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 334 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 335 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 336 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 337 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 338 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 339 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 340 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 341 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 342 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 343 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 344 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 345 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 346 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 347 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 348 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 349 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 350 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 351 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 352 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 353 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 354 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 499 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "txcmpl_0",
+						  "txcmpl_1",
+						  "txcmpl_2",
+						  "txcmpl_3",
+						  "txcmpl_4",
+						  "txcmpl_5",
+						  "txcmpl_6",
+						  "txcmpl_7",
+						  "txcmpl_8",
+						  "txcmpl_9",
+						  "txcmpl_10",
+						  "txcmpl_11",
+						  "txcmpl_12",
+						  "txcmpl_13",
+						  "txcmpl_14",
+						  "txcmpl_15",
+						  "txcmpl_16",
+						  "txcmpl_17",
+						  "txcmpl_18",
+						  "txcmpl_19",
+						  "txcmpl_20",
+						  "txcmpl_21",
+						  "txcmpl_22",
+						  "txcmpl_23",
+						  "txcmpl_24",
+						  "txcmpl_25",
+						  "txcmpl_26",
+						  "txcmpl_27",
+						  "txcmpl_28",
+						  "txcmpl_29",
+						  "txcmpl_30",
+						  "txcmpl_31",
+						  "rxfill_0",
+						  "rxfill_1",
+						  "rxfill_2",
+						  "rxfill_3",
+						  "rxfill_4",
+						  "rxfill_5",
+						  "rxfill_6",
+						  "rxfill_7",
+						  "rxdesc_0",
+						  "rxdesc_1",
+						  "rxdesc_2",
+						  "rxdesc_3",
+						  "rxdesc_4",
+						  "rxdesc_5",
+						  "rxdesc_6",
+						  "rxdesc_7",
+						  "rxdesc_8",
+						  "rxdesc_9",
+						  "rxdesc_10",
+						  "rxdesc_11",
+						  "rxdesc_12",
+						  "rxdesc_13",
+						  "rxdesc_14",
+						  "rxdesc_15",
+						  "rxdesc_16",
+						  "rxdesc_17",
+						  "rxdesc_18",
+						  "rxdesc_19",
+						  "rxdesc_20",
+						  "rxdesc_21",
+						  "rxdesc_22",
+						  "rxdesc_23",
+						  "misc";
+			};
 		};
 
 		pcs0: ethernet-pcs@7a00000 {
